docs/compactor.md

```diff
@@ -0,0 +1,168 @@
+
+## Solution
+### Compaction command
+To help with the problem mentioned earlier, our proposal is to introduce `compact` subcommand with `etcdbrctl`. On execution of `compact` command, A separate embedded ETCD process will be started  where the ETCD data will be restored from the snapstore (exactly as in the restoration scenario today). Then the new ETCD database will be compacted and defragmented using ETCD API calls. The compaction will strip off the ETCD database of old revisions as per the ETCD auto-compaction configuration. The defragmentation will free up the unused fragment memory space released after compaction. Then a full snapshot of the compacted database will be saved in snapstore which then can be used as the base snapshot during any subsequent restoration (or backup compaction).
+
+### How the solution works
+The newly introduced compact command does not disturb the running ETCD while compacting the backup snapshots. The command is designed to run potentially separately (from the main ETCD process/container/pod). ETCD Druid may take care of starting the newly introduced compact command as a separate job (scheduled periodically) based on a parameter which is defined in ETCD CRD as  `backupCompactionSchedule`.
+
+### Example ETCD CRD: 
+```yaml
+apiVersion: druid.gardener.cloud/v1alpha1
+kind: Etcd
+...
+spec:
+  ...
+  backup:
+    ...
+    backupCompactionSchedule: "0/30 * * * *" # Backup compaction job is executed every 30m on the clock
+    ...
+  ...
+  ```
+
+  ### Example Cron Job:
+  ```yaml
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+...
+spec:
+  schedule: {{ .Values.backup.backupCompactionSchedule }} 
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      backoffLimit: 0
+      template:
+        metadata:
+        ...
+        spec:
+          restartPolicy: Never
+          containers:
+          - name: compact-backup
+            image: {{ .Values.backup.image }} 
+            imagePullPolicy: {{ .Values.backup.pullPolicy }} 
+            command:
+            - etcdbrctl
+            - compact
+            - --data-dir=/var/etcd/data/new.etcd
+{{- if  .Values.store.storageProvider }}
+            - --storage-provider={{ .Values.store.storageProvider }}
+{{- end }}
+{{- if .Values.store.storePrefix }}
+            - --store-prefix={{ .Values.store.storePrefix }}
+{{- end }}
+{{- if .Values.backup.etcdQuotaBytes }}
+            - --embedded-etcd-quota-bytes={{ int $.Values.backup.etcdQuotaBytes }}
+{{- end }}
+{{- if .Values.store.storageContainer }}
+            - --store-container={{ .Values.store.storageContainer }}
+{{- end }}
+            ports:
+            - containerPort: {{ .Values.backup.port }}
+              name: server
+              protocol: TCP
+            resources:
+{{ toYaml .Values.backup.resources | indent 14 }}
```

Comment:
Giving helm template content in the example doesn't make sense. Could you please clean the variable substitution part up. Or alternatively, just hyperlink to the corresponding template file in the helm chart in the repo. Then you can skip giving the `CronJob` example. The `Etcd` CRD example looks good.